### PR TITLE
Enhance Logs readability of  SDK Conformance Tests

### PR DESCRIPTION
**Pull request by aLekSer**
Add more detaied logging for local SDK server. Make `npm` and `apt-get` less noisy. `apt-get install openjdk-8-jre` spares about 25% of output.

For #1452 .

### CODE REVIEW

File: build/build-sdk-images/node/build-sdk-test.sh [hunk 1]
```diff
@@ -15,8 +15,8 @@
 # limitations under the License.
 
 cd /go/src/agones.dev/agones/test/sdk/nodejs
-npm install
-npm rebuild
+npm install --quiet
```
Comment (markmandel) [inline build/build-sdk-images/node/build-sdk-test.sh:18]: Nice!

File: build/build-sdk-images/restapi/Dockerfile [hunk 2]
```diff
@@ -26,10 +26,12 @@ ENV GOPATH /go
 RUN wget -q https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz && \
     tar -xzf go${GO_VERSION}.linux-amd64.tar.gz && rm go${GO_VERSION}.linux-amd64.tar.gz && mkdir -p ${GOPATH}
 
+RUN echo '§' && apt-get -qy update 
 RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
-    apt-get install -y nodejs
+    apt-get install -qq -y nodejs
```
Comment (markmandel) [inline build/build-sdk-images/restapi/Dockerfile:31]: Also nice! Good idea to reduce the noise!

File: build/includes/sdk.mk [hunk 3]
```diff
@@ -143,7 +143,7 @@ run-sdk-conformance-no-build: FEATURE_GATES ?=
 run-sdk-conformance-no-build: ensure-agones-sdk-image
 run-sdk-conformance-no-build: ensure-build-sdk-image
 	DOCKER_RUN_ARGS="--net host -e AGONES_SDK_GRPC_PORT=$(GRPC_PORT) -e AGONES_SDK_HTTP_PORT=$(HTTP_PORT) -e FEATURE_GATES=$(FEATURE_GATES) $(DOCKER_RUN_ARGS)" COMMAND=sdktest $(MAKE) run-sdk-command & \
-	docker run -p $(GRPC_PORT):$(GRPC_PORT) -p $(HTTP_PORT):$(HTTP_PORT) -e "FEATURE_GATES=$(FEATURE_GATES)" -e "ADDRESS=" -e "TEST=$(TESTS)" -e "TIMEOUT=$(TIMEOUT)" -e "DELAY=$(DELAY)" \
+	docker run -p $(GRPC_PORT):$(GRPC_PORT) -p $(HTTP_PORT):$(HTTP_PORT) -e "FEATURE_GATES=$(FEATURE_GATES)" -e "ADDRESS=" -e "TEST=$(TESTS)" -e "SDK_NAME=$(SDK_FOLDER)" -e "TIMEOUT=$(TIMEOUT)" -e "DELAY=$(DELAY)" \
```
Comment (markmandel) [inline build/includes/sdk.mk:146]: :raised_hands:

File: pkg/sdkserver/localsdk.go [hunk 4]
```diff
@@ -92,8 +94,10 @@ func NewLocalSDKServer(filePath string) (*LocalSDKServer, error) {
 		updateObservers: sync.Map{},
 		requestSequence: make([]string, 0),
 		testMode:        false,
+		testSdkName:     "",
 		gsState:         agonesv1.GameServerStateScheduled,
 	}
+	l.logger = runtime.NewLoggerWithType(l)
```
Comment (markmandel) [inline pkg/sdkserver/localsdk.go:100]: Nice addition!

File: pkg/sdkserver/localsdk.go [hunk 5]
```diff
@@ -161,11 +165,20 @@ func (l *LocalSDKServer) SetExpectedSequence(sequence []string) {
 	l.expectedSequence = sequence
 }
 
+// SetSdkName set SDK name to be added to the logs
+func (l *LocalSDKServer) SetSdkName(sdkName string) {
+	l.testSdkName = sdkName
+	l.logger = l.logger.WithField("sdkName", l.testSdkName)
```
Comment (markmandel) [inline pkg/sdkserver/localsdk.go:171]: Perfect. Love it.

Comment (aLekSer) [reply pkg/sdkserver/localsdk.go:171]: Thanks now at least logs from local SDK server would be labeled with SDK name. Hope to label SDK test clients in the future (second half of logs).

**Comment by aLekSer** [comment]
Updated a master, no chages to the code.

File: pkg/sdkserver/localsdk.go [hunk 6]
```diff
@@ -163,13 +167,22 @@ func (l *LocalSDKServer) SetExpectedSequence(sequence []string) {
...
+
 // recordRequest append request name to slice
 func (l *LocalSDKServer) recordRequest(request string) {
 	if l.testMode {
 		l.testMutex.Lock()
 		defer l.testMutex.Unlock()
 		l.requestSequence = append(l.requestSequence, request)
 	}
+	if l.testSdkName != "" {
+		l.logger.Debugf("Received %s requets", request)
```
Comment (akremsa) [inline pkg/sdkserver/localsdk.go:184]: Typo: request

File: pkg/sdkserver/localsdk.go [hunk 7]
```diff
@@ -443,42 +456,44 @@ func (l *LocalSDKServer) Close() {
...
 		if _, ok := aSet[v]; !ok {
+			l.logger.Infof("Found an element which was not expected %s", v)
 			return false
 		}
 		bSet[v] = true
 	}
-	for _, v := range a {
+	for _, v := range expected {
 		if _, ok := bSet[v]; !ok {
+			l.logger.Infof("Could not find an element which was expected %s", v)
```
Comment (akremsa) [inline pkg/sdkserver/localsdk.go:476]: logger.WithError().   ?

File: pkg/sdkserver/localsdk.go [hunk 8]
```diff
@@ -443,42 +456,44 @@ func (l *LocalSDKServer) Close() {
...
 	aSet := make(map[string]bool)
 	bSet := make(map[string]bool)
-	for _, v := range a {
+	for _, v := range expected {
 		aSet[v] = true
 	}
-	for _, v := range b {
+	for _, v := range received {
 		if _, ok := aSet[v]; !ok {
+			l.logger.Infof("Found an element which was not expected %s", v)
```
Comment (akremsa) [inline pkg/sdkserver/localsdk.go:469]: .WithField().Error() ?

Comment (aLekSer) [reply pkg/sdkserver/localsdk.go]: Thanks, here definitely should go an error, as well as this helps me to identify one `fmt.Print` usage.

**Pull request merged**